src/engine.h

```diff
@@ -96,10 +96,10 @@ class EngineController {
   RpSharedMutex busy_mutex_;
   using SharedLock = std::shared_lock<RpSharedMutex>;
 
-  std::unique_ptr<Search> search_;
   std::unique_ptr<NodeTree> tree_;
   std::unique_ptr<SyzygyTablebase> syzygy_tb_;
   std::unique_ptr<Network> network_;
+  std::unique_ptr<Search> search_;
```

Comment:
Also after `cache_`. Maybe better to have `cache_` first in the list for symmetry.

Reply:
```cpp
// Note that the order is important here. Search has to be destructed before
// variables it depends on (e.g. tree_, syzygy_tb_, network_ and cache_), so it
// should be listed last.
```